src/AzureExtension/DataManager/AzureDataManager.cs

```diff
@@ -560,64 +608,330 @@ private async Task UpdateDataForPullRequestsAsync(DataStoreOperationParameters p
                 case PullRequestView.Unknown:
                     throw new ArgumentException("PullRequestView is unknown");
                 case PullRequestView.Mine:
-                    searchCriteria.CreatorId = result.Connection!.AuthorizedIdentity.Id;
+                    searchCriteria.CreatorId = connectionResult.Connection!.AuthorizedIdentity.Id;
                     break;
                 case PullRequestView.Assigned:
-                    searchCriteria.ReviewerId = result.Connection!.AuthorizedIdentity.Id;
+                    searchCriteria.ReviewerId = connectionResult.Connection!.AuthorizedIdentity.Id;
                     break;
                 case PullRequestView.All:
                     /* Nothing different for this */
                     break;
             }
 
-            var pullRequests = await gitClient.GetPullRequestsAsync(project.InternalId, azureUri.Repository, searchCriteria, null, null, PullRequestResultLimit);
-            if (pullRequests == null || pullRequests.Count == 0)
+            var pullRequests = await gitClient.GetPullRequestsAsync(project.InternalId, repository.InternalId, searchCriteria, null, null, PullRequestResultLimit);
+            await ProcessPullRequests(
+                connectionResult.Connection!,
+                pullRequests,
+                project,
+                repository,
+                parameters.DeveloperId.LoginId,
+                parameters.PullRequestView,
+                parameters.OperationName == "UpdateDataForDeveloperPullRequestsAsync");
+        } // Foreach AzureUri
+
+        return;
+    }
+
+    public async Task UpdatePullRequestsForLoggedInDeveloperIdsAsync(RequestOptions? options = null, Guid? requestor = null)

```

Comment:
NIT: Guid.Empty instead of null.

Reply:
The guid is generated if it is null, not empty. And is invalid to have a non-constant default value, so it is null and the method sets it to Guid.NewGuid() if it is null.